ledger/acctupdates.go

```diff
@@ -376,60 +523,535 @@ func (au *accountUpdates) getAssetCreatorForRound(rnd basics.Round, aidx basics.
 	return au.accountsq.lookupAssetCreator(aidx)
 }
 
-func (au *accountUpdates) committedUpTo(rnd basics.Round) basics.Round {
+// committedUpTo enqueues commiting the balances for round committedRound-lookback.
+// The defered committing is done so that we could calculate the historical balances lookback rounds back.
+// Since we don't want to hold off the tracker's mutex for too long, we'll defer the database persistance of this
+// operation to a syncer goroutine. The one caviat is that when storing a catchpoint round, we would want to
+// wait until the catchpoint creation is done, so that the persistance of the catchpoint file would have an
+// uninterrupted view of the balances at a given point of time.
+func (au *accountUpdates) committedUpTo(committedRound basics.Round) (retRound basics.Round) {
+	var isCatchpointRound, hasMultipleIntermediateCatchpoint bool
+	var offset uint64
+	var dc deferedCommit
+	au.accountsMu.RLock()
+	defer func() {
+		au.accountsMu.RUnlock()
+		if dc.offset != 0 {
```

Comment:
exactly. But we also want to write to the channel only after releasing the lock, and I'd rather release the lock once instead of releasing it, writing to the channel and taking the lock again.